client/state/themes/utils.js

```diff
@@ -2,25 +2,47 @@
  * External dependencies
  */
 import startsWith from 'lodash/startsWith';
-import { filter, get, map, mapKeys, omit, omitBy, split } from 'lodash';
+import {
+	every,
+	filter,
+	get,
+	includes,
+	map,
+	mapKeys,
+	omit,
+	omitBy,
+	some,
+	split
+} from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { isThemeMatchingQuery } from 'lib/query-manager/theme/util';
 import { DEFAULT_THEME_QUERY } from './constants';
 
 /**
  * Constants
  */
 const REGEXP_SERIALIZED_QUERY = /^(?:(\d+):)?(.*)$/;
+const SEARCH_TAXONOMIES = [ 'subject', 'feature', 'color', 'style', 'column', 'layout' ];
```

Comment:
This is not the full set of filter taxonomies. It should probably be pulled from the [object](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/themes/theme-filters.js#L26) here if we really need to know about it. I'm worried about this though, because filtering should be completely transparent to the theme state.